CSharp/core-DirectLine/README.md

```diff
@@ -51,49 +51,51 @@ while (true)
     {
         if (input.Length > 0)
         {
-            Message userMessage = new Message
+            Activity userMessage = new Activity
             {
-                FromProperty = fromUser,
-                Text = input
+                From = new ChannelAccount(fromUser),
+                Text = input,
+                Type = ActivityTypes.Message
             };
 
-            await client.Conversations.PostMessageAsync(conversation.ConversationId, userMessage);
+            await client.Conversations.PostActivityAsync(conversation.ConversationId, userMessage);
         }
     }
 }
 ````
 
-Messages from the Bot are continually polled from the API in another Thread in the `ReadBotMessagesAsync` method. Check out the [Program.cs](DirectLineClient/Program.cs#L64-L69) usage of `GetMessagesAsync` method which retrieves conversation messages newer than the stored watermark. Messages are then filtered to receive messages from the Bot only.
+Messages from the Bot are continually polled from the API in another Thread in the `ReadBotMessagesAsync` method. Check out the [Program.cs](DirectLineClient/Program.cs#L64-L69) usage of `GetActivitiesAsync` method which retrieves conversation messages newer than the stored watermark. Activities are then filtered to receive messages from the Bot only.
 
 ````C#
-var messages = await client.Conversations.GetMessagesAsync(conversationId, watermark);
-watermark = messages?.Watermark;
+var activitySet = await client.Conversations.GetActivitiesAsync(conversationId, watermark);
+watermark = activitySet?.Watermark;
 
-var messagesFromBotText = from x in messages.Messages
-                    where x.FromProperty == botId
+var activities = from x in activitySet.Activities
+                    where x.From.Id == botId
                     select x;
 ````
 
-The ChannelData property provides a way for you to send native metadata to take advantage of special features or concepts for a channel (see [Custom Channel Capabilities](https://docs.botframework.com/en-us/csharp/builder/sdkreference/channels.html) for more information.) Check out the `ReadBotMessagesAsync` method in [Program.cs](DirectLineClient/Program.cs#L75-L91) to where the ChannelData custom content is deserialized and render appropriately.
+DirectLine v3.0 (unlike version 1.1) has supports for Attachments (see [Adding Attachments to a Message](https://docs.botframework.com/en-us/core-concepts/attachments) for more information about attachments). Check out the `ReadBotMessagesAsync` method in [Program.cs](DirectLineClient/Program.cs#L75-L92) to see where the Attachments are retrieved and rendered appropriately based on their type.
```

Comment:
*...) to see **where** the Attachment are retrieved and rendered...*
shouldn't it read?
*...) to see **how** the Attachment are retrieved and rendered...*